### PR TITLE
SMHE-2076 - secret managment embedded tomcat

### PR DESCRIPTION
**Pull request by jboon**
Reconfigured secret-management to create a springboot jar with embedded tomcat

### CODE REVIEW

**Comment by Marcel-Jansen** [comment]
https://jenkins.fdp.osgp.cloud/view/SMHE-cucumber/job/gxf-apps-smart-metering-integration-smhe-cucumber-build/934/

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/resources/osgp-adapter-protocol-dlms.properties [hunk 1]
```diff
@@ -3,7 +3,7 @@ soapclient.max-conn-total=100
 soapclient.use.client.auth=true
 soapclient.use.hostname.verifier=false
 soapclient.supported.tls.protocols=TLSv1.2,TLSv1.3
-soapclient.default-uri=https://localhost:443/osgp-secret-management/ws/SecretManagement
```
Comment (Marcel-Jansen) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/resources/osgp-adapter-protocol-dlms.properties:6]: Waarom stappen we hier af van HTTPS?

**Review by Marcel-Jansen — CHANGES_REQUESTED** [review]

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/resources/osgp-adapter-protocol-dlms.properties (hunk 1 above)
Comment (jboon) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/resources/osgp-adapter-protocol-dlms.properties:6]: fixed

File: osgp/platform/osgp-secret-management/src/main/resources/logback-spring.xml [hunk 2]
```diff
@@ -7,42 +7,7 @@ SPDX-License-Identifier: Apache-2.0
 -->
 
 <configuration>
-
-  <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
-    <encoder>
-      <pattern>[%d{yyyy-MM-dd HH:mm:ss.SSS}] [${HOSTNAME}] [%thread] %level %logger{36}@%method:%line - %msg%n</pattern>
```
Comment (Marcel-Jansen) [inline osgp/platform/osgp-secret-management/src/main/resources/logback-spring.xml:13]: Blijft dit logging pattern gehandhaaft?
`[%d{yyyy-MM-dd HH:mm:ss.SSS}] [${HOSTNAME}] [%thread] %level %logger{36}@%method:%line - %msg%n`

Comment (Marcel-Jansen) [inline osgp/platform/osgp-secret-management/src/main/resources/logback-spring.xml:13]: Blijft dit logging pattern gehandhaaft?
`[%d{yyyy-MM-dd HH:mm:ss.SSS}] [${HOSTNAME}] [%thread] %level %logger{36}@%method:%line - %msg%n`

File: osgp/platform/osgp-secret-management/src/main/resources/logback-spring.xml [hunk 3]
```diff
@@ -7,42 +7,7 @@ SPDX-License-Identifier: Apache-2.0
...
-  <logger name="org.opensmartgridplatform.shared.security.providers" level="DEBUG"/>
-
-  <logger name="org.opensmartgridplatform.adapter.ws.endpointinterceptors" level="TRACE"/>
-  <logger name="org.opensmartgridplatform.adapter.protocol.dlms.domain.commands" level="INFO"/>
-  <logger name="org.opensmartgridplatform.adapter.ws.shared.services" level="INFO"/>
-
-  <logger name="org.springframework.ws.server.MessageTracing" level="TRACE" />
-
+  <include resource="org/springframework/boot/logging/logback/base.xml"/>
   <logger name="org.opensmartgridplatform.secretmanagement.application" level="DEBUG"/>
```
Comment (Marcel-Jansen) [inline osgp/platform/osgp-secret-management/src/main/resources/logback-spring.xml]: By default zijn we alleen geïnteresseerd in INFO logging:
```
<root level="INFO" additivity="false">
  <appender-ref ref="CONSOLE"/>
</root>
```

**Review by Marcel-Jansen — CHANGES_REQUESTED** [review]

File: osgp/platform/osgp-secret-management/src/main/resources/logback-spring.xml (hunk 3 above)
Comment (jboon) [reply osgp/platform/osgp-secret-management/src/main/resources/logback-spring.xml]: removed debug level: levels can be configured in application.properties or overruled by external log file defined in environment variable LOGGING_CONFIG

File: osgp/platform/osgp-secret-management/src/main/resources/logback-spring.xml (hunk 2 above)
Comment (jboon) [reply osgp/platform/osgp-secret-management/src/main/resources/logback-spring.xml:13]: logging pattern added in application.properties

**Review by Marcel-Jansen — DISMISSED** [review]

File: osgp/platform/osgp-secret-management/pom.xml [hunk 4]
```diff
@@ -23,14 +22,12 @@ SPDX-License-Identifier: Apache-2.0
...
-    <jaxb.plugin.version>2.5.0</jaxb.plugin.version>
-    <build.helper.maven.plugin.version>3.0.0</build.helper.maven.plugin.version>
+    <osgp-ws-secret-management-version>${project.parent.version}</osgp-ws-secret-management-version>
     <apache.ws.xlmschema.version>2.3.1</apache.ws.xlmschema.version>
     <wsdl4j.version>1.6.3</wsdl4j.version>
-    <maven.failsafe.plugin.version>3.2.5</maven.failsafe.plugin.version>
-    <osgp-ws-secret-management-version>${project.parent.version}</osgp-ws-secret-management-version>
     <h2database.version>2.2.224</h2database.version>
+    <build.helper.maven.plugin.version>3.5.0</build.helper.maven.plugin.version>
+    <cyclonedx-maven-plugin.version>2.7.11</cyclonedx-maven-plugin.version>
```
Comment (harrymiddelburgetall) [inline osgp/platform/osgp-secret-management/pom.xml]: Version in super pom is 2.7.2. Why not upgrade there?

**Review by stefanermens — DISMISSED** [review]

File: osgp/platform/osgp-secret-management/pom.xml (hunk 4 above)
Comment (jboon) [reply osgp/platform/osgp-secret-management/pom.xml]: done

**Review by harrymiddelburgetall — APPROVED** [review]

**Review by Marcel-Jansen — APPROVED** [review]

**Pull request merged**